### PR TITLE
Add Single of own PackageInfo. Implements #12

### PR DESCRIPTION
**Pull request by dmfs**
This also adds two Singles and one Iterable to retrieve version and permission data from a given PackageInfo.

### CODE REVIEW

**Comment by dmfs** [comment]
rebased on master

File: package-bolts/src/main/java/org/dmfs/android/bolts/apppackage/AppPackageInfo.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+ *
+ * @author Marten Gajda
+ */
+public final class AppPackageInfo implements Single<PackageInfo>
+{
+    private final Context mContext;
+    private final int mFlags;
+
+
+    public AppPackageInfo(Context context, int flags)
```
Comment (lemonboston) [inline package-bolts/src/main/java/org/dmfs/android/bolts/apppackage/AppPackageInfo.java:38]: Maybe a secondary ctor with default flag 0?

File: package-bolts/src/main/java/org/dmfs/android/bolts/apppackage/RequestedAppPermissions.java [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+import org.dmfs.iterators.elementary.Seq;
+import org.dmfs.jems.single.Single;
+import org.dmfs.optional.NullSafe;
+import org.dmfs.optional.decorators.Mapped;
+
+import java.util.Iterator;
+
+
+/**
+ * An {@link Iterable} of all permissions declared in the app manifest using a {@code <uses-permission>} element. This includes permissions which have not be
```
Comment (lemonboston) [inline package-bolts/src/main/java/org/dmfs/android/bolts/apppackage/RequestedAppPermissions.java]: just a typo: "haven not be" -> "been"

**Review by lemonboston — APPROVED** [review]
Looks good to me.

File: package-bolts/src/main/java/org/dmfs/android/bolts/apppackage/AppPackageInfo.java (hunk 1 above)
Comment (dmfs) [reply package-bolts/src/main/java/org/dmfs/android/bolts/apppackage/AppPackageInfo.java:38]: If you're forced to provide flags it's harder to forget them when necessary.

File: package-bolts/src/main/java/org/dmfs/android/bolts/apppackage/RequestedAppPermissions.java (hunk 2 above)
Comment (dmfs) [reply package-bolts/src/main/java/org/dmfs/android/bolts/apppackage/RequestedAppPermissions.java]: fixed.

File: package-bolts/src/main/java/org/dmfs/android/bolts/apppackage/AppPackageInfo.java (hunk 1 above)
Comment (lemonboston) [reply package-bolts/src/main/java/org/dmfs/android/bolts/apppackage/AppPackageInfo.java:38]: ok

**Review by lemonboston — APPROVED** [review]

**Pull request merged**